src/main/java/Decide.java

```diff
@@ -175,32 +205,33 @@ public static boolean LIC4(Point[] pts, Parameters params) {
     }
 
     /**
-     * Checks for LIC3 in the input parameters.
+     * Checks if there is two points, separated by K_pts consecutive intervening
+     * points, with a distance greater than LENGTH1 between each other.
      *
-     * LIC3 is true if the input contains three consecutive points that forms a
-     * triangle with an area larger than AREA1.
-     * @param pts array of Points objects
-     * @param params Parameter object
-     * @return true if LIC3 as described above is true, false otherwise.
+     * @param pts array of data points of type Point
+     * @param params Parameter object acting as configuration
+     * @return boolean which is true iff LIC7 condition is met
      */
-    public static boolean LIC3(Point[] pts, Parameters params) {
-        if(params.area1 < 0) {
-            throw new IllegalArgumentException("AREA1 must be greater or equal to 0");
+    public static boolean LIC7(Point[] pts, Parameters params) {
+        if (1 <= params.k_pts && params.k_pts <= pts.length) { /* Condition fulfilled */
```

Comment:
Should be "params.k_pts <= pts.length **- 2**"

Reply:
Really good catch! Due to this i also discovered that the order should be changed, am I right?
Since NUMPOINT is guaranteed to be >= 2 and k_points is required to be >= 1. This means that (k_pts=1) <= (NUMPOINTS=2) - 2 would throw an exception, but it is a valid input.

Reply:
Yes that's right, good point 👍